src/components/maskedInput/new.tsx

```diff
@@ -0,0 +1,129 @@
+import React, {
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+  useImperativeHandle,
+  forwardRef,
+  ReactElement,
+  ForwardedRef
+} from 'react';
+import _ from 'lodash';
+import {StyleSheet, Keyboard, TextInput, TextInputProps, StyleProp, ViewStyle} from 'react-native';
+import View from '../view';
+import Text from '../text';
+import TouchableOpacity from '../touchableOpacity';
+
+export interface MaskedInputProps extends TextInputProps {
+  /**
+   * Initial value to pass to masked input
+   */
+  initialValue?: string;
+  /**
+   * callback for rendering the custom input out of the value returns from the actual input
+   */
+  renderMaskedText?: ReactElement;
+  /**
+   * Custom formatter for the input value
+   */
+  formatter?: (value?: string) => string | undefined;
+  /**
+   * container style for the masked input container
+   */
+  containerStyle: StyleProp<ViewStyle>;
+}
+
+/**
+ * @description: Mask Input to create custom looking inputs with custom formats
+ * @gif: https://camo.githubusercontent.com/61eedb65e968845d5eac713dcd21a69691571fb1/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f4b5a5a7446666f486f454b334b2f67697068792e676966
+ * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/MaskedInputScreen.js
+ */
+function MaskedInput(props: MaskedInputProps, ref: ForwardedRef<any>) {
```

Comment:
I've written this in the main PR area, they should make that more noticeable somehow...
> Consider moving the screen to TS as well.

Reply:
Definitely. Fixed and also add a new cool example (: